src/nvim/testdir/test_assert.vim

```diff
@@ -1,5 +1,11 @@
 " Test that the methods used for testing work.
 
+func Test_assert_fails_in_try_block()
+  try
+    call assert_equal(0, assert_fails('throw "error"'))
```

Comment:
8.0.1770 is not merged so this shouldn't work even if the test passes.

Reply:
I've checked it.. :)

Without 8.0.1770 it returns 0 - without the patch the test fails.

Reply:
If `assert_fails` returns a non-zero value after 8.0.1770, then this patch failed the test. `assert_fails` always returns 0 before 8.0.1770.

Reply:
But the test fails without the/this patch:

> function RunTheTest[37]..Test_assert_fails_in_try_block line 2: command did not fail: throw "error"

8.0.1770 changes behavior in this regard, but it is not really important for this patch/test.